File: tyama_icpc2007rA.vb
module icpc2007rA
	sub Main(ByVal args() as String)
		dim n,k,m,r,i as integer
		while true
			dim s(3) as string
			s=Split(Console.ReadLine())
			n=s(0):k=s(1):m=s(2):r=0
			if n=0
				exit while
			end if
			for i=1 to n-1
				r=(r+k) mod i
			next
			Console.WriteLine((r+m) mod n + 1)
		end while
	end sub
end module

File: tyama_icpc2007rA.fs
open System
let mutable arg:String array=Console.ReadLine().Split(' ')
let mutable n=int(arg.[0])
while n>0 do
 let mutable r=0
 let k=int(arg.[1])
 let m=int(arg.[2])
 for i in 1..n-1 do
  r<-(r+k)%i
 Console.WriteLine((r+m)%n+1)
 arg<-Console.ReadLine().Split(' ')
 n<-int(arg.[0])